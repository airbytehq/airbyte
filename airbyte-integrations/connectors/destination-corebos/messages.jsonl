
{"type": "RECORD", "record": {"stream": "DatasetItems", "data": {"ticket_title": "About product not working", "from_mailscanner": "0", "parent_id": "12x1638", "assigned_user_id": "19x11", "product_id": "14x2631", "ticketpriorities": "Normal", "ticketstatus": "Open", "ticketseverities": "Major", "hours": "45.000000", "createdtime": "2015-04-07 16:20:12", "ticketcategories": "Other Problem", "days": "1", "update_log": "Ticket created. Assigned to  user  Administrator -- Monday 15th February 2016 12:01:29 AM by admin--//--", "modifiedtime": "2016-06-27 13:27:25", "ticket_no": "TT00096", "from_portal": "0", "modifiedby": "19x1", "email": "tu_rivet@gmail.com", "commentadded": "", "created_user_id": "19x1", "description": " Gloriosa ostentatio in constituendo summo bono. Hoc ille tuus non vult omnibusque ex rebus voluptatem quasi mercedem exigit. Heri, inquam, ludis commissis ex urbe profectus veni ad vesperum. Itaque hic ipse iam pridem est reiectus; Duo Reges: constructio interrete. De illis, cum volemus. Venit enim mihi Platonis in mentem, quem accepimus primum hic disputare solitum; Non minor, inquit, voluptas percipitur ex vilissimis rebus quam ex pretiosissimis. \n\n", "solution": "", "id": "17x2731"}, "emitted_at": 1636402938000}}

{"stream": "DatasetItems", "data": {"ticket_title": "About product not working", "from_mailscanner": "0", "parent_id": "12x1638", "assigned_user_id": "19x11", "product_id": "14x2631", "ticketpriorities": "Normal", "ticketstatus": "Open", "ticketseverities": "Major", "hours": "45.000000", "createdtime": "2015-04-07 16:20:12", "ticketcategories": "Other Problem", "days": "1", "update_log": "Ticket created. Assigned to  user  Administrator -- Monday 15th February 2016 12:01:29 AM by admin--//--", "modifiedtime": "2016-06-27 13:27:25", "ticket_no": "TT00096", "from_portal": "0", "modifiedby": "19x1", "email": "tu_rivet@gmail.com", "commentadded": "", "created_user_id": "19x1", "description": " Gloriosa ostentatio in constituendo summo bono. Hoc ille tuus non vult omnibusque ex rebus voluptatem quasi mercedem exigit. Heri, inquam, ludis commissis ex urbe profectus veni ad vesperum. Itaque hic ipse iam pridem est reiectus; Duo Reges: constructio interrete. De illis, cum volemus. Venit enim mihi Platonis in mentem, quem accepimus primum hic disputare solitum; Non minor, inquit, voluptas percipitur ex vilissimis rebus quam ex pretiosissimis. \n\n", "solution": "", "id": "17x2731"}, "emitted_at": 1636402938000}