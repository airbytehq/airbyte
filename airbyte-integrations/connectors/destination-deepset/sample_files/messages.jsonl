{"stream": "test_unstructured", "data": {"content": "# Heading\n\nThis is the content which is not just a single word", "document_key": "testdoc_pdf.pdf", "_ab_source_file_last_modified": "2023-10-27T00:45:58.000000Z", "_ab_source_file_url": "testdoc_pdf.pdf", "_ab_source_file_parse_error": null}, "emitted_at": 1698400264556}
{"stream": "test_unstructured", "data": {"content": "# Heading\n\nThis is the content which is not just a single word", "document_key": "testdoc_google", "_ab_source_file_last_modified": "2023-11-10T13:46:18.551000Z", "_ab_source_file_url": "testdoc_google", "_ab_source_file_parse_error": null}, "emitted_at": 1698400261074}
{"stream": "test_unstructured", "data": {"content": "# A Simple PDF File\n\nThis is a small demonstration .pdf file -\n\njust for use in the Virtual Mechanics tutorials. More text. And more text. And more text. And more text. And more text.\n\nAnd more text. And more text. And more text. And more text. And more text. And more text. Boring, zzzzz. And more text. And more text. And more text. And more text. And more text. And more text. And more text. And more text. And more text.\n\nAnd more text. And more text. And more text. And more text. And more text. And more text. And more text. Even more. Continued on page 2 ...\n\n# Simple PDF File 2\n\n...continued from page 1. Yet more text. And more text. And more text. And more text. And more text. And more text. And more text. And more text. Oh, how boring typing this stuff. But not as boring as watching paint dry. And more text. And more text. And more text. And more text. Boring. More, a little more text. The end, and just as well.", "document_key": "https://airbyte179.sharepoint.com/Shared%20Documents/Test_folder/Test_foler_2_1/simple_pdf_file.pdf", "_ab_source_file_parse_error": null, "_ab_source_file_last_modified": "2023-12-15T16:47:21.000000Z", "_ab_source_file_url": "https://airbyte179.sharepoint.com/Shared%20Documents/Test_folder/Test_foler_2_1/simple_pdf_file.pdf"}, "emitted_at": 1710365942797}
