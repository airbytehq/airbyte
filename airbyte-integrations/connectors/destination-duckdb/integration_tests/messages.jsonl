{"type": "RECORD", "record": {"stream": "airbyte_acceptance_table", "emitted_at": 1664705198575, "data": { "column1": "test", "column2": 222, "column3": "2022-06-20T18:56:18", "column4": 33.33, "column5": [1,2,null]}}}
