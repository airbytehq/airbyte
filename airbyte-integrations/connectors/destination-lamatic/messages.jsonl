{"type": "RECORD", "record": {"stream": "test", "emitted_at": 1664705198575, "data": { "id": 12, "name": "Vansh"}}}
{"type": "RECORD", "record": {"stream": "test", "emitted_at": 1664705198575, "data": { "id": 16, "name": "Mehul"}}}