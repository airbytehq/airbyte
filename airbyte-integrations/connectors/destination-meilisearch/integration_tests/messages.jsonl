{"type": "RECORD", "record": {"stream": "ab-airbyte-testing", "data": {"_ab_pk": "my_value", "column2": 221, "column3": "2021-01-01T20:10:22", "column4": 1.214, "column5": [1,2,3]}, "emitted_at": 1626172757000}}
{"type": "RECORD", "record": {"stream": "ab-airbyte-testing", "data": {"_ab_pk": "my_value2", "column2": 222, "column3": "2021-01-02T22:10:22", "column5": [1,2,null]}, "emitted_at": 1626172757000}}
