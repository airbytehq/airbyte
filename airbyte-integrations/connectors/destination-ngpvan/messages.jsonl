{"type": "RECORD", "record": {"stream": "organizations", "data": {"id": 1, "modified_date": 1638557221, "created_date": 1511902789, "name": "Mobilize", "is_primary_campaign": false, "candidate_name": "", "slug": "mobilize", "is_coordinated": false, "state": "", "race_type": null, "district": "", "logo_url": "https://mobilize-uploads-prod.s3.us-east-2.amazonaws.com/uploads/organization/logo-for-mobilize-feed%403x_20200930173319016754.png", "org_type": "CAMPAIGN", "is_independent": false, "event_feed_url": "https://www.mobilize.us/"}, "emitted_at": 1643403855888}}
{"type": "RECORD", "record": {"stream": "organizations", "data": {"id": 2, "modified_date": 1604938057, "created_date": 1511902898, "name": "31st St Swing Left", "is_primary_campaign": false, "candidate_name": "", "slug": "31st-st-swing-left", "is_coordinated": true, "state": "", "race_type": null, "district": "", "logo_url": "", "org_type": "COORDINATED", "is_independent": false, "event_feed_url": "https://www.mobilize.us/31st-st-swing-left/"}, "emitted_at": 1643403855892}}