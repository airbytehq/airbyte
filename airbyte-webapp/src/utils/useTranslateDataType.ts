import { useMemo } from "react";
import { useIntl } from "react-intl";

export interface AirbyteConnectorData {
  type: string;
  $ref?: string;
  format?: string;
  airbyte_type?: string;
  anyOf?: unknown[];
  oneOf?: unknown[];
}

const wellKnownTypesPrefix = "WellKnownTypes.json#/definitions/";
const wellKnownTypeToDataTypeKey = {
  [`${wellKnownTypesPrefix}String`]: "string",
  [`${wellKnownTypesPrefix}BinaryData`]: "binary_data",
  [`${wellKnownTypesPrefix}Number`]: "number",
  [`${wellKnownTypesPrefix}Integer`]: "integer",
  [`${wellKnownTypesPrefix}Boolean`]: "boolean",
  [`${wellKnownTypesPrefix}Date`]: "date",
  [`${wellKnownTypesPrefix}TimestampWithTimezone`]: "timestamp_with_timezone",
  [`${wellKnownTypesPrefix}TimestampWithoutTimezone`]: "timestamp_without_timezone",
  [`${wellKnownTypesPrefix}TimeWithTimezone`]: "time_with_timezone",
  [`${wellKnownTypesPrefix}TimeWithoutTimezone`]: "time_without_timezone",
};

const getType = (data: AirbyteConnectorData): string => {
  if (data.oneOf || data.anyOf) {
    return "union";
  }
  if (!data.anyOf && !data.oneOf && !data.airbyte_type && !data.format && !data.type && !data.$ref) {
    return "unknown";
  }
  if (data.$ref) {
    return wellKnownTypeToDataTypeKey[data.$ref] ?? "unknown";
  }
  return data.airbyte_type ?? data.format ?? data.type;
};

export const useTranslateDataType = (props: AirbyteConnectorData) => {
  const { formatMessage } = useIntl();
  const dataType = useMemo(() => getType(props), [props]);
  return useMemo(
    () =>
      formatMessage({
        id: `airbyte.datatype.${dataType}`,
        defaultMessage: formatMessage({ id: "airbyte.datatype.unknown" }),
      }),
    [dataType, formatMessage]
  );
};

// another ver of useTranslateDataType - without hook and return just string. The old one is left for backward compatibility
export const getDataType = (data: AirbyteConnectorData): string => {
  return `airbyte.datatype.${getType(data)}`;
};
