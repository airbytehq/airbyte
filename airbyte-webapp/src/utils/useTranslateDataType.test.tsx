import { renderHook } from "@testing-library/react-hooks";
import React from "react";
import { IntlProvider } from "react-intl";

import { AirbyteConnectorData, useTranslateDataType } from "./useTranslateDataType";
import messages from "../locales/en.json";

const wrapper: React.FC<React.PropsWithChildren<unknown>> = ({ children }) => (
  <IntlProvider messages={messages} locale="en">
    {children}
  </IntlProvider>
);
const translate = (translatable: AirbyteConnectorData) => {
  const {
    result: { current },
  } = renderHook(() => useTranslateDataType(translatable), { wrapper });
  return current;
};

describe("useTranslateDataType", () => {
  it.each([
    "string",
    "date",
    "timestamp_with_timezone",
    "timestamp_without_timezone",
    "datetime",
    "integer",
    "big_integer",
    "number",
    "big_number",
    "array",
    "object",
    "untyped",
    "union",
    "boolean",
  ])("should translate %", (type) => {
    const current = translate({ type });
    expect(current).toMatchSnapshot();
    expect(current).not.toBe(type);
  });

  describe("union type", () => {
    it("should use union type if oneOf is set", () => {
      expect(translate({ type: "string", oneOf: [] })).toBe("Union");
    });
    it("should use union type if anyOf is set", () => {
      expect(translate({ type: "string", anyOf: [] })).toBe("Union");
    });
    it("should use union type if anyOf and oneOf is set", () => {
      expect(translate({ type: "string", anyOf: [], oneOf: [] })).toBe("Union");
    });
  });

  it("should be unknown if the type is not recognized", () => {
    expect(translate({ type: "potato" })).toBe("Unknown");
  });

  it("should return untyped when given no values", () => {
    // @ts-expect-error Using this method internally should demand a type
    expect(translate({})).toBe("Unknown");
  });

  describe("translation priority", () => {
    it("should use airbyte_type over format or type", () => {
      expect(translate({ type: "string", format: "date", airbyte_type: "big_number" })).toBe("Big Number");
    });
    it("should use format over type", () => {
      expect(translate({ type: "string", format: "date" })).toBe("Date");
    });
    it("should use oneOf/anyOf over everything else", () => {
      expect(translate({ type: "string", format: "date", airbyte_type: "big_number", oneOf: [] })).toBe("Union");
    });
  });
});
