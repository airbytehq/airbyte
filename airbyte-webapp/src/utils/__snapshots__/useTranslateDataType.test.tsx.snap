// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useTranslateDataType should translate % 1`] = `"String"`;

exports[`useTranslateDataType should translate % 2`] = `"Date"`;

exports[`useTranslateDataType should translate % 3`] = `"Timestamp with Timezone"`;

exports[`useTranslateDataType should translate % 4`] = `"Datetime"`;

exports[`useTranslateDataType should translate % 5`] = `"Datetime"`;

exports[`useTranslateDataType should translate % 6`] = `"Integer"`;

exports[`useTranslateDataType should translate % 7`] = `"Big Integer"`;

exports[`useTranslateDataType should translate % 8`] = `"Number"`;

exports[`useTranslateDataType should translate % 9`] = `"Big Number"`;

exports[`useTranslateDataType should translate % 10`] = `"Array"`;

exports[`useTranslateDataType should translate % 11`] = `"Object"`;

exports[`useTranslateDataType should translate % 12`] = `"Untyped"`;

exports[`useTranslateDataType should translate % 13`] = `"Union"`;

exports[`useTranslateDataType should translate % 14`] = `"Boolean"`;
